frontend/src/metabase/ui/components/theme/ThemeProvider/ThemeProvider.unit.spec.tsx

```diff
@@ -0,0 +1,45 @@
+import type { MantineThemeOverride } from "@mantine/core";
+import { render, screen } from "@testing-library/react";
+
+import { getThemeOverrides } from "metabase/ui/theme";
+
+import { Text } from "../..";
+
+import { ThemeProvider } from "./ThemeProvider";
+
+describe("ThemeProvider", () => {
+  const OVERRIDES = getThemeOverrides();
+
+  it("applies the metabase theme overrides", () => {
+    render(
+      <ThemeProvider>
+        <Text size="xl">Text</Text>
+      </ThemeProvider>,
+    );
+
+    const textCss = window.getComputedStyle(screen.getByText("Text"));
+    expect(textCss.getPropertyValue("font-size")).toBe(OVERRIDES.fontSizes!.xl);
+  });
+
+  it("merges the theme overrides when the theme prop is provided", () => {
+    // Simulate a user-provided theme override.
+    // This is primarily used by the React embedding SDK.
+    const theme: MantineThemeOverride = {
+      colors: { "text-dark": ["rgb(12, 34, 56)"] },
+    };
+
+    render(
+      <ThemeProvider theme={theme}>
+        <Text size="xl">Text</Text>
+      </ThemeProvider>,
+    );
+
+    const textCss = window.getComputedStyle(screen.getByText("Text"));
+
+    // Metabase theme overrides should still be preserved.
+    expect(textCss.getPropertyValue("font-size")).toBe(OVERRIDES.fontSizes!.xl);
+
+    // Theme overrides from the user should be applied.
+    expect(textCss.getPropertyValue("color")).toBe("rgb(12, 34, 56)");
```

Comment:
I think we should also assert in the previous test that the color would be from the default theme. Whatever `color` value would be. That way this assertion would be more obvious that it overrides the default value.

Reply:
@WiNloSt good idea! I added that test case 👍🏻